### ISSUE TITLE
How to use TabPFN as a module in a larger model

### ISSUE THREAD

**Issue by tyf0416** [PriorLabs/TabPFN#702]
### Describe the workflow you want to enable

Hello TabPFN community,

I am trying to apply the pretrained TabPFN as a block inside a larger model, with the following flow structure. The Encoder and Decoder will be learned duruing backpropagation, while the TabPFN is fixed and the gradient information passes through TabPFN.

**Input --> Encoder --> TabPFN --> Decoder --> Output**

In such case, the **TabPFN.get_preprocessed_datasets()** gets the tensor matrix from Encoder as the input containing gradient information. However, current function only accepts Numpy as the input for preprocessing. I am wondering whether there is any quick solution for this case, enventhough it was mentioned in the function **initialize_dataset_preprocessing** that the differentiable input is not supported for regressors yet.

### Describe your proposed solution

A new pipeline to allow TabPFN to work as a part in the larger modela and pass the gradient computation graph

### Impact

None

**Comment by noahho** [comment]
This would indeed be quite helpful! This is on our stack but we currently don't know when this feature could be ready unfortunetaly.

**Comment by mabynke** [comment]
I'm interested to know if this is in the works. It would be very useful!
